src/plugins/discover/public/application/view_components/utils/use_search.ts

```diff
@@ -161,7 +164,6 @@ export const useSearch = (services: DiscoverViewServices) => {
     dataset = searchSource.getField('index');
 
     let elapsedMs;
-
     try {
       // Only show loading indicator if we are fetching when the rows are empty
       if (fetchStateRef.current.rows?.length === 0) {
```

Comment:
if `fetchStateRef.current` has no rows, `undefined === 0` will be false. I think what you really want is
```suggestion
      if (!(fetchStateRef.current.rows?.length)) {
```

Reply:
this PR did not change this line. 

but i think we want the if statement here only check the case: `length === 0`